### ISSUE TITLE
Proposed install tutorial with Docker

### ISSUE THREAD

**Issue by szat** [YihangChen-ee/HAC-plus#8]
Dear authors,

First of all thank you for the amazing paper and strong results! I was playing with the installation, and on a more modern gpu (RTX4900) for me this Docker isolation strategy worked. Maybe it is useful for others :)

**Step 1: Clean Your Environment**
Remove broken containers or dangling images to ensure a fresh build.
Bash
docker rm -f hac_container

Optional: remove the old image to force a 100% fresh build
docker rmi hac-plus-new 

**Step 2: The Final Dockerfile**
Save this as Dockerfile in /home/szatmari/Code/3DGS_HAC-plus. It automates the installation of the tmc3 codec, fixes the Intel MKL iJIT error, and automatically activates the environment for you.
Dockerfile
```
FROM nvidia/cuda:11.6.2-devel-ubuntu20.04

ENV DEBIAN_FRONTEND=noninteractive
ENV CONDA_DIR=/opt/conda
ENV PATH=$CONDA_DIR/bin:$PATH

# 1. System Dependencies
RUN apt-get update && apt-get install -y \
    wget \
    git \
    build-essential \
    cmake \
    libtiff5 \
    && rm -rf /var/lib/apt/lists/*

# 2. Compile TMC3 (MPEG-PCC-TMC13) Codec from source
WORKDIR /opt
RUN git clone https://github.com/MPEGGroup/mpeg-pcc-tmc13.git && \
    cd mpeg-pcc-tmc13 && \
    mkdir build && \
    cd build && \
    cmake .. && \
    make -j$(nproc)
ENV PATH="/opt/mpeg-pcc-tmc13/build/tmc3:${PATH}"

# 3. Install Miniconda
RUN wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh && \
    /bin/bash ~/miniconda.sh -b -p /opt/conda && \
    rm ~/miniconda.sh

# 4. Conda Environment Setup
RUN conda tos accept --override-channels --channel https://repo.anaconda.com/pkgs/main && \
    conda tos accept --override-channels --channel https://repo.anaconda.com/pkgs/r

# mkl==2021.4.0 prevents the iJIT_NotifyEvent crash.
RUN conda create -n HAC_env -y -c pytorch -c pyg -c conda-forge -c defaults \
    python=3.7.13 \
    pytorch=1.12.1 \
    torchvision=0.13.1 \
    torchaudio=0.12.1 \
    cudatoolkit=11.6 \
    plyfile=0.8.1 \
    pip=22.3.1 \
    pytorch-scatter \
    mkl==2021.4.0 \
    tqdm

# Switch execution context to the Conda environment
SHELL ["conda", "run", "-n", "HAC_env", "/bin/bash", "-c"]
RUN pip install einops wandb lpips

# 5. Compile Custom CUDA Submodules
COPY submodules/ /tmp/submodules/
ENV TORCH_CUDA_ARCH_LIST="7.0;7.5;8.0;8.6+PTX"

RUN cd /tmp/submodules/diff-gaussian-rasterization && pip install . && \
    cd /tmp/submodules/simple-knn && pip install . && \
    cd /tmp/submodules/gridencoder && pip install . && \
    cd /tmp/submodules/arithmetic && pip install .

RUN rm -rf /tmp/submodules

# 6. Automate Environment Activation
RUN conda init bash
RUN echo "conda activate HAC_env" >> /root/.bashrc

WORKDIR /app
```

**Step 3: Build the Image**
This compiles the C++ code for multiple architectures.
Bash
```
cd ~/Code/3DGS_HAC-plus
docker build --network=host -t hac-plus-new .
```

**Step 4: Create and Launch the Container**
Bash
```
docker run --gpus all -it \
    --name hac_container \
    --network host \
    -v ~/Code/3DGS_HAC-plus:/app \
    -v ~/Data/static_input_3dgs/blending:/app/data/static_input_3dgs/blending \
    -v ~/Data/static_output_3dgs/hacpp_output:/app/outputs/blending \
    hac-plus-new /bin/bash
```

**Step 5: Patch the PyTorch Source Code**
On your host machine, open ~/Code/3DGS_HAC-plus/train.py at line 180.
Change: scaling_reg = scaling.prod(dim=1).mean() To: scaling_reg = (scaling[:, 0] * scaling[:, 1] * scaling[:, 2]).mean()

**Step 6: Create the Training Script**
Save as run_training.py in ~/Code/3DGS_HAC-plus.
Python
```
import os
os.environ["PYTORCH_NVFUSER_DISABLE"] = "1"
os.environ["PYTORCH_JIT"] = "0"

for lmbda in [0.004, 0.001]:
    for cuda, scene in enumerate(['playroom', 'drjohnson']):
        mask_lr_final = 0.00008 * lmbda / 0.001
        one_cmd = f'CUDA_VISIBLE_DEVICES={0} python train.py -s data/static_input_3dgs/blending/{scene} --eval --lod 0 --voxel_size 0.005 --update_init_factor 16 --iterations 100 -m outputs/blending/{scene}/{lmbda} --lmbda {lmbda} --mask_lr_final {mask_lr_final} --test_iterations 100 --save_iterations 100'
        print(f"Executing: {one_cmd}")
        os.system(one_cmd)

output_dir = 'outputs/blending'
if os.path.exists(output_dir):
    os.system(f'chmod -R 777 {output_dir}')

```
**Step 7: Execute Training**
Inside the container terminal (it will already be activated):
Bash
`python run_training.py`

**Step 8: Pause and Resume**
To pause, type exit. To resume from host:
Bash
```
docker start hac_container
docker exec -it hac_container bash
# No need to activate conda; it's automatic.
python run_training.py

```
**Step 9: Changing Mapped Folders**
To change host paths without rebuilding:
    1. docker rm -f hac_container
    2. Re-run Step 4 with the updated host paths in the -v flags.
